### ISSUE TITLE
How to use map and map_clamp's eq parameter?

### ISSUE THREAD

**Issue by tqwewe** [nannou-org/nannou#840]
Nannou has two functions [`map`](https://docs.rs/nannou/latest/nannou/ease/fn.map.html) and [`map_clamp`](https://docs.rs/nannou/latest/nannou/ease/fn.map_clamp.html) which are very useful.. but I'm having a hard time understanding what should be put in the last parameter `eq: impl FnOnce(T, T, T, T) -> T`.

Could someone provide guidance on this? I think the docs could also do with an example.

**Comment by tqwewe** [comment]
It seems like the last parameter expects an easing function... but pennereq or nannou don't even seem to provide a linear easing function 🤦 

To manually use, you can do:
```rust
clamp(num, 0, 10, 0, 100, |t, b, c, d| c * t / d + b);
```

**Comment by sidwellr** [comment]
I agree the docs could be better! I think you're right about the last parameter being an easing function; after all `map` and `map_clamp` are in the `ease` module. For linear easing, you can use `map_range`, which is included in the Nannou prelude.

**Comment by tqwewe** [comment]
Ah it seems like `map_range` is exactly what I was after. It seems like map_range does the same thing as `map` but with linear easing.

I think the docs should probably point users to `map_range` to help clarify this.

**Comment by PaulDotSH** [comment]
> Ah it seems like `map_range` is exactly what I was after. It seems like map_range does the same thing as `map` but with linear easing.
> 
> I think the docs should probably point users to `map_range` to help clarify this.

How did you manage to use map_range since it only takes 2 arguments?
Could you provide an example?

i.e. what i have atm is
ease::map(star.x/star.y,0.,1.,0.,bounds.0 as f32, map_range);

**Comment by tqwewe** [comment]
> How did you manage to use map_range since it only takes 2 arguments?

`map_range` takes 5 arguments - https://docs.rs/nannou/latest/nannou/math/fn.map_range.html

The documentation has some examples on how to use it.
In your code snippet, can you just do:
```rust
nannou::math::map_range(star.x / star.y, 0.0, 1.0, 0.0, bounds.0 as f32);
```

**Comment by PaulDotSH** [comment]
> > How did you manage to use map_range since it only takes 2 arguments?
> 
> `map_range` takes 5 arguments - https://docs.rs/nannou/latest/nannou/math/fn.map_range.html
> 
> The documentation has some examples on how to use it. In your code snippet, can you just do:
> 
> ```rust
> nannou::math::map_range(star.x / star.y, 0.0, 1.0, 0.0, bounds.0 as f32);
> ```

Thanks a lot, however with the bounds being 512, I manage to get values like x: 0 y: 3069424000000000000000000000000, is this normal behaviour?

**Comment by tqwewe** [comment]
It should be fine, I don't think there's any issues with the function. It all depends on the inputs. If you're casting, make sure nothing overflows

**Comment by PaulDotSH** [comment]
> It should be fine, I don't think there's any issues with the function. It all depends on the inputs. If you're casting, make sure nothing overflows

Yeah, the problem was that the input wasnt in the input bounds given to the function
